src/app/session/pages/validation/validation.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SessionService } from '../../services/session.service';
+
+@Component({
+  selector: 'app-validation',
+  templateUrl: './validation.component.html',
+  styleUrls: ['./validation.component.css'],
+})
+export class ValidationComponent implements OnInit {
+
+  private token: string;
+
+  constructor(private route: ActivatedRoute, private sessionService: SessionService, private router: Router) {}
+
+  ngOnInit() {
+    this.route.queryParams.subscribe((params) => {
```

Comment:
use a routeSnaphot instead of putting subscribe in subscribe.
Never use subscribe in subscribe